enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,439 @@
+
+Configuring dracut is done via [kernel command line
+arguments](https://mirrors.edge.kernel.org/pub/linux/utils/boot/dracut/dracut.html#_network).
+This can be customized via PXE configuration or at the grub command line during
+boot.
+
+Networking at this stage of the installation process is important, as the
+installer generates a “pointer ignition config”, or a small ignition stub that
+just points to a URL for where to download the full ignition configuration from
+the Machine Config Server.  In other words, when a new RHCOS host boots for the
+first time, dracut must configure networking just enough for ignition to be
+able to reach the Machine Config Server.
+
+A [past bug against
+dracut](https://bugzilla.redhat.com/show_bug.cgi?id=1787620) ensured that there
+was a way to get dracut to attempt both DHCPv4 (IPv4) and DHCPv6 (IPv6) and not
+fail if one of them worked.  We also [changed
+OpenShift](https://bugzilla.redhat.com/show_bug.cgi?id=1793591) to use the
+`ip=dhcp,dhcp6` configuration by default in RHCOS.  This at least got default
+behavior far enough along that it will work in either an IPv4 or IPv6
+environment.
+
+Assuming dracut succeeds in at least one DHCP attempt, additional networking
+can be configured via ignition if further customizations are needed to enable
+the host to reach the Machine Config Server.
+
+### Ignition
+
+[Ignition](https://github.com/coreos/ignition/) is the tool used by RHCOS to do
+initial host configuration on first boot.  It runs in initramfs, so the
+networking set up done by dracut is the network environment ignition runs in.
+
+Currently there are two ignition configuration files for each role e.g
+master/worker - a “pointer” ignition is created by the installer, and
+this can be customized by the user via the installer `create ignition-configs`
+interface.  This configuration then references the rendered configuration via
+an ignition `append` directive, which downloads the configuration referenced
+from the Machine Config Server (which requires network access).
+
+Here is an example of a “pointer” ignition configuration as generated by the
+OpenShift installer:
+
+```
+$ cat ocp/ostest/worker.ign
+{"ignition":{"config":{"append":[{"source":"https://[fd2e:6f44:5dd8:c956::5]:22623/config/worker","verification":{}}]},"security":{"tls":{"certificateAuthorities":[{"source":"data:text/plain;charset=utf-8;base64,...","verification":{}}]}},"timeouts":{},"version":"2.2.0"},"networkd":{},"passwd":{},"storage":{},"systemd":{}}
+```
+
+By customizing the “pointer” ignition configuration generated by the installer,
+additional network configuration can be done on the host prior to attempting to
+reach the Machine Config Server URL.
+
+#### Example
+
+Consider a cluster with hosts that have many network interfaces.  We may want
+to ensure that a set of these interfaces is completely disabled. First we
+create NetworkManager configuration files that disable the set of desired
+interfaces.
```

Comment:
perhaps a better approach is for us to document how to change NetworkManager to default to not attempt to bring up networking on interfaces that don't have explicit configuration.. Via Ignition you'd create a file like:

```
$ cat /etc/NetworkManager/conf.d/10-disable-unconfigured-nics.conf
[main]
no-auto-default=*
```

Reply:
That's a good idea.  I actually got this example from the docs for a Red Hat lab that was using exactly this example, so I knew it worked at least.  What you suggest here is more elegant.  I can put that in a comment back on those lab docs and tag you.

Reply:
i've applied this to the doc